### ISSUE TITLE
copyArtifacts from lastStable instead of lastSuccessful

### ISSUE THREAD

**Issue by Hejado** [eclipse/xtext#1512]
*Background:*
Jenkins has a weird terminology problem about what build results actually mean. When evaluating the build result in a post stage, success means stable (green build only):

So, when we use archiveArtifacts like this:

```
post{
  success {
    // not executed when unstable
    archiveArtifacts artifacts: 'build/**'
  }
}
```

we won't end up with artifacts if tests fail - which is by itself okay.

The sign-and-deploy job, however, tries to copy the artifacts using the lastSuccessful qualifier. Here, Jenkins weirdly ignores the distinction between unstable and successful. `lastSuccessful` points to green and yellow builds alike. 

So when sign-and-deploy tries to retrieve the upstream artifacts, and one or more builds were unstable, it fails because lastSuccessful points to a build without artifacts: https://ci-staging.eclipse.org/xtext/job/releng/job/sign-and-deploy/108/console

*in short:*
Whenever we copy artifacts from another job, we should use the lastStable quailfier instead, as this would resolve this issue and we probably don't want to release unstable artifacts anyway.

**Comment by Hejado** [comment]
Done.
